### DISCUSSION TITLE
SPs and clients should be able to mark content as not available for retrieval

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/lotus#8522] [General Q&A]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [X] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

Currently, all deals are being indexed and this would allow anyone in the world to request the retrieval of private data.
Some the SPs have private data deals with their client would prefer if these could not be retrieved by other others.

### Describe the solution you'd like

A discussion around this has already taken place under https://github.com/filecoin-project/go-fil-markets/issues/704.
I will leave the implementation logic upto the devs as there are 2 thought process.
1. Don't index the deal but allow retrieval (need change to deal proposal only)
2. Don't index the deal and don't allow retrieval except for client (need change to deal proposal - not sure about logic to allow client)

### Describe alternatives you've considered

Data can be simply archived with passwords and thus would be safe. But this makes retrievals complicated for large dataset.

### Additional context

Boost already has a similar issue https://github.com/filecoin-project/boost/issues/455. Please keep this request open till completion or official deprecation of go-fil-market.

**Comment by jennijuju** [comment]
"should have this feature" is controversial -  as on some level, filecoin is a public network. currently, there is no way for clients to enforce un-retreivability for the content they stored with SP, tho there might be future FIP or user smart contract to deliver this feature. if a client desire such feature, they should encrypt the data before sending it atm.

Im converting this to discussion for collecting ideas.
